### DISCUSSION TITLE
Max and custom steps on Y axis

### DISCUSSION THREAD

**Discussion by DannyBloky** [patrykandpatrick/vico#517] [Questions]
Hi I don't know if it already exist but didn't see it.

I now have implement your chart. 

I use number but convert them to floats. 
But I got large numbers.

But I don't know how to set a max to the y Axis. Because I don't want my largest point to be at the the top.

Also now I get random numbers als label like 240345 ect. 
So is it possible to do like steps like 20000 or something depends on your data. Because that looks nicer

Just some suggestion how it should work 

I would like to say that my yAxis must go to 900000(900k)
I want to do it in 5 steps of 200000(200k)

So my yAxis would show these numbers: 0, 200k, 400k, 600k, 800k (and then full number or I need to add my own converter to show 200k instead of 200000)
So the 800k will not be at the top because I want it to be 900k

Maybe this is not possible yet. But I would be great if this can be implemented in the future

For other people this wil also be nice I think. And probably also on the x axis

**Comment by christianrowlands** [comment]
Greetings @DannyBloky. I am not a maintainer or developer on this project, so I can't fully answer your question, but just yesterday I starting using this library and had a similar need. I set the max to the y Axis using the following code...

Note: I am using the 2.0.0-alpha.5 version, so it might be different if you are using the 1.X.X version.

```kotlin
private val axisValueOverrider = AxisValueOverrider.fixed<LineCartesianLayerModel>(
    maxY = -30f,
    minY = -110f,
    maxX = 70f,
    minX = 10f,
)
```

and then I use that value in the Host

```kotlin
        CartesianChartHost(
            chart =
            rememberCartesianChart(
                rememberLineCartesianLayer(
                    axisValueOverrider = axisValueOverrider,
                    lines = remember(defaultLines) { defaultLines.map { it.copy(backgroundShader = null) } },
                ),
                fadingEdges = rememberFadingEdges(),
            ),
            modelProducer = modelProducer,
            marker = rememberMarker(),
            runInitialAnimation = false,
            diffAnimationSpec = snap(),
            horizontalLayout = horizontalLayout,
            chartScrollSpec = rememberChartScrollSpec(isScrollEnabled = false)
        )
```

As for the yAxis labels, I have not figured that part out yet, but will comment when I do.

**Reply by christianrowlands** [reply]
@DannyBloky , I was able to limit the number of yAxis labels using the `itemPlacer`. I set mine to 5, and that turned out to be even intervals and set the values to -30, -50, -70, -90, and -110. You might also need to do what was mentioned in this discussion: https://github.com/patrykandpatrick/vico/discussions/514#discussioncomment-7975830

```kotlin
startAxis = rememberStartAxis(
                    guideline = null,
                    horizontalLabelPosition = VerticalAxis.HorizontalLabelPosition.Inside,
                    itemPlacer = remember { AxisItemPlacer.Vertical.default({ _ -> 5 }) },
                ),
```

**Comment by DannyBloky** [comment]
Thanks for the comment.
I'm using version 1.13.1 and using the xml

I'm using this chart:
` <com.patrykandpatrick.vico.views.chart.ChartView
            android:id="@+id/lineChart"
            android:layout_width="match_parent"
            android:layout_height="200dp"
            app:chart="line"
            app:showBottomAxis="true"
            app:showStartAxis="true" />`
            
            
So for people that do that also this is the fix for it:

` viewModel.roundedYValue.observe(viewLifecycleOwner) { value ->
            binding.lineChart.chart?.axisValuesOverrider = AxisValuesOverrider.fixed(minY = 0f, maxY = value.toFloat())
        }`
        
I use to float because it I have a list of Int's

I'm using this to get the largest point and from there I set my steps.

`
fun getRoundedYValue(numbers: List<Int>): Int {

        val maxValue = numbers.maxOrNull() ?: 0
        return when {
            maxValue < 10 -> 10
            maxValue < 20 -> 20
            maxValue < 30 -> 30
            maxValue < 40 -> 40
            maxValue < 50 -> 50
            maxValue < 75 -> 75
            maxValue < 100 -> 100
            maxValue < 1000 -> ((maxValue + 100 - 1) / 100) * 100
            maxValue < 10000 -> ((maxValue + 1000 - 1) / 1000) * 1000
            maxValue < 100000 -> ((maxValue + 10000 - 1) / 10000) * 10000
            else -> ((maxValue + 100000 - 1) / 100000) * 100000
        }
    }`

**Comment by patrickmichalik** [comment]
Hello! As mentioned by @christianrowlands, you can customize the value range of an axis via `AxisValueOverrider` (`AxisValuesOverrider` in Vico 1), and you can use `AxisItemPlacer.Vertical.default` to change the number of labels along a _y_-axis. If you’d like your _y_-axis to have a range of 0 to 900K and labels for 0, 200K, 400K, and 800K, you can create a custom `AxisItemPlacer.Vertical` implementation. (We plan on adding such a feature to the default implementation in the future. You’ll be able to specify either the label count or the step, much like in Google Sheets.) To format your labels as described (“200K,” “400K,” and so on), you can use `AxisValueFormatter`. Let me know if you have any questions. Thanks for the help, @christianrowlands!
